### PR TITLE
Allow to set type of Popconfirm OK button

### PR DESCRIPTION
**Pull request by yociduo**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [x] Update API docs for the component.
  * [x] Update/Add demo to demonstrate new feature.
  * [x] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

### CODE REVIEW

**Comment by yociduo** [comment]
Link Issue #6840

File: components/popconfirm/index.en-US.md [hunk 1]
```diff
@@ -20,7 +20,9 @@ The difference with `confirm` is more lightweight than the static popped full-sc
 | onConfirm | callback of confirmation                           | function(e)      | none     |
 | onCancel  | callback of cancel                           | function(e)      | none     |
 | okText    | text of the confirmation button                              | string        | Confirm   |
+| okType    | type of the confirmation button                              | string        | primary   |
 | cancelText| text of the cancel button                              | string        | Cancel   |
+| cancelType| type of the confirmation button                              | string        | none   |
```
Comment (afc163) [inline components/popconfirm/index.en-US.md]: I don't think we need a `cancelType`, let's keep it as simple as possible.

**Comment by afc163** [comment]
`Modal.confirm` would need this feature too.

**Comment by yociduo** [comment]
`Modal.info` need the feature? @afc163

**Comment by afc163** [comment]
Nope.

**Comment by yociduo** [comment]
@afc163 Added

**Comment by afc163** [comment]
Could you rebase the `feature-2.13`?

**Comment by yociduo** [comment]
Ok

**Comment by afc163** [comment]
Please make sure travis ci is passed~

**Comment by yociduo** [comment]
Passed

**Comment by benjycui** [comment]
> @afc163 : I don't think we need a cancelType, let's keep it as simple as possible. 

@yociduo

File: components/popconfirm/index.en-US.md (hunk 1 above)
Comment (afc163) [inline components/popconfirm/index.en-US.md]: Remove `cancelType`.

**Comment by yociduo** [comment]
@benjycui  remove cancel type, but I think cancel type is useful, because people can't override the footer in `Popconfirm`

**Comment by afc163** [comment]
I don't think we need another type of cancel type but `default` type.

**Comment by yociduo** [comment]
Ok, removed

File: components/popconfirm/demo/basic.md [hunk 2]
```diff
@@ -27,7 +27,7 @@ function cancel(e) {
 }
 
 ReactDOM.render(
-  <Popconfirm title="Are you sure delete this task?" onConfirm={confirm} onCancel={cancel} okText="Yes" cancelText="No">
+  <Popconfirm title="Are you sure delete this task?" onConfirm={confirm} onCancel={cancel} okText="Yes" okType="danger" cancelText="No" cancelType="dashed">
```
Comment (benjycui) [inline components/popconfirm/demo/basic.md]: You forget to update demo

File: components/modal/index.en-US.md [hunk 3]
```diff
@@ -65,7 +66,9 @@ The properties of the object are follows:
 | width      | Width of dialog           | string\|number | 416           |
 | iconType   | Type of Icon component    | string | question-circle |
 | okText     | Text of OK button    | string           | OK       |
+| okType    | type of the confirmation button                              | string        | primary   |
 | cancelText | Text of cancel button    | string           | Cancel       |
+| cancelType| type of the confirmation button                              | string        | none   |
```
Comment (benjycui) [inline components/modal/index.en-US.md]: And docs ....

Just `grep -r 'cancelType' ./components` to find all the cancelType and remove it.

Comment (yociduo) [inline components/modal/index.en-US.md]: Sorry, updated

File: components/popconfirm/demo/basic.md [hunk 4]
```diff
@@ -27,7 +27,7 @@ function cancel(e) {
 }
 
 ReactDOM.render(
-  <Popconfirm title="Are you sure delete this task?" onConfirm={confirm} onCancel={cancel} okText="Yes" cancelText="No">
+  <Popconfirm title="Are you sure delete this task?" onConfirm={confirm} onCancel={cancel} okText="Yes" okType="danger" cancelText="No">
```
Comment (afc163) [inline components/popconfirm/demo/basic.md]: 这个例子不要改。

**Review by benjycui — APPROVED** [review]

**Pull request merged**